src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/TextBasedPhysicalPlanPrinterImpl2.java

```diff
@@ -0,0 +1,334 @@
+		}
+	}
+
+	protected void printOperatorInfoForOpLocalToGlobal ( final PhysicalOpLocalToGlobal op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "l2g (" + op.getID() + ") " );
+		final LogicalOperator lop = op.getLogicalOperator();
+		printLogicalOperator( nameOfLogicalOp(lop), lop.getID(), out, indentLevelStringForOpDetail );
+		out.append( indentLevelStringForOpDetail + singleBase + "  - vocab.mapping " + ((LogicalOpLocalToGlobal) lop).getVocabularyMapping().hashCode() );
+		out.append( System.lineSeparator() );
+	}
+	
+	protected void printOperatorInfoForMultiwayUnion ( final PhysicalOpMultiwayUnion op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "multiwayUnion (" + op.getID() + ") " );
+		printLogicalOperator( nameOfLogicalOp(op.getLogicalOperator()), op.getLogicalOperator().getID(), out, indentLevelStringForOpDetail );
+	}
+
+	protected void printOperatorInfoForOpNaiveNestedLoopsJoin ( final PhysicalOpNaiveNestedLoopsJoin op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "naiveNestedLoop (" + op.getID() + ") " );
+		printLogicalOperator( nameOfLogicalOp(op.getLogicalOperator()), op.getLogicalOperator().getID(), out, indentLevelStringForOpDetail );
+	}
+
+	protected void printOperatorInfoForOpParallelMultiLeftJoin ( final PhysicalOpParallelMultiLeftJoin op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "parallelMultiLeftJoin (" + op.getID() + ") " );
+		out.append( System.lineSeparator() );
+	}
+	
+	protected void printOperatorInfoForRequest( final PhysicalOpRequest op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		final LogicalOpRequest lop = op.getLogicalOperator();
+		final DataRetrievalRequest req = lop.getRequest();
+		out.append( indentLevelString + "req (" + op.getID() + ")" );
+		out.append( System.lineSeparator() );
+		printFederationMember( lop.getFederationMember(), indentLevelStringForOpDetail, out );
+		out.append( indentLevelStringForOpDetail + "  - pattern (" + req.hashCode() +  ") (" + req.toString() + ")" );
+		out.append( System.lineSeparator() );
+	}
+	
+	protected void printOperatorInfoForOpSymmetricHashJoin ( final PhysicalOpSymmetricHashJoin op, final PrintStream out, final String indentLevelString, final String indentLevelStringForOpDetail ) {
+		out.append( indentLevelString + "symmetricHashJoin (" + op.getID() + ") " );
+		printLogicalOperator( nameOfLogicalOp(op.getLogicalOperator()), op.getLogicalOperator().getID(), out, indentLevelStringForOpDetail );
+	}
+	
+	protected String nameOfLogicalOp ( final LogicalOperator lop) {
```

Comment:
You can move this function to the base class and, then, use it in the plan printer for logical plans as well. This way we will be guaranteed to have consistency regarding the short names used for the logical operators.